java-squid/src/main/java/org/sonar/java/resolve/SemanticModel.java

```diff
@@ -137,19 +120,14 @@ public Tree getTree(Resolve.Env env) {
   }
 
   public Resolve.Env getEnv(Tree tree) {
-    JavaTree javaTree = (JavaTree) tree;
     Resolve.Env result = null;
-    while (result == null && javaTree != null) {
-      result = envs.get(javaTree);
-      javaTree = (JavaTree) parentLink.get(javaTree);
+    while (result == null && tree != null) {
+      result = envs.get(tree);
+      tree = tree.parent();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "tree". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)